### PR TITLE
Add dockerComposeRun step

### PR DESCRIPTION
**Pull request by giorgiosironi**
To simplify https://github.com/elifesciences/sciencebeam-alignment/pull/2

### CODE REVIEW

File: vars/dockerComposeRun.groovy [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import DockerCompose
+
+def call(service, command, tag='latest')
+{
+    try {
+        sh DockerCompose
+            .command('run', ['docker-compose.yml', 'docker-compose.ci.yml'])
```
Comment (de-code) [inline vars/dockerComposeRun.groovy]: we don't need the `-f` here?

**Comment by de-code** [comment]
There is one case where we get the output (version) using `returnStdout: true`, could we add support for that as well? (or is there already support for that?)

File: vars/dockerComposeRun.groovy (hunk 1 above)
Comment (giorgiosironi) [reply vars/dockerComposeRun.groovy]: it's added by https://github.com/elifesciences/elife-jenkins-workflow-libs/blob/master/src/DockerCompose.groovy#L53-L55

**Comment by giorgiosironi** [comment]
Yes, we should return the output, it can still be ignored by the caller if not needed.

**Comment by de-code** [comment]
I guess there are cases where we care more about the status than the output (I believe to have seen that somewhere). Might be worth to use a parameter instead?

**Comment by giorgiosironi** [comment]
Multiple return types (as `sh` does) can be confusing, so we can add a separate step for the special cases. In general a step that has a `returnStatus != 0` is failing, but it's usually retried (and we should deal with the retries here too) or propagated up the stack. Not sure if you can get the `returnStatus` from the exception.

**Comment by de-code** [comment]
Actually the same Jenkinsfile is using it to check whether the tag exists (I copied that from [proofreader](https://github.com/elifesciences/proofreader-python/blob/387b3c65ee7777e26b3a7340179dc4ed68f24f58/Jenkinsfile#L19)). Although not docker compose in that case.
adding `returnStdout` as a parameter might also make it more explicit that it is returning the output?

**Comment by giorgiosironi** [comment]
Discussion on getting some introspection like the exit code out of the exception: https://issues.jenkins-ci.org/browse/JENKINS-28822

If just an `sh` needs to be run, no need to run a `dockerCompose*` custom step instead; that can be customized through its options. The intent is only to remove the boilerplate to run `docker-compose -f ... run $service $command` which was executed multiple times with different variations.

**Comment by de-code** [comment]
Okay, my preference would be to add an explicit `returnStdout` parameter to make it more obvious that we do return the output and if we ever needed returnStatus we could add it in the future too. No need to add `returnStatus` now.

**Comment by giorgiosironi** [comment]
Groovy parameters are not as amenable as Python ones (cannot be re-ordered or passed by name easily unless the parameter is defined as a generic `Map`), so we may need a separate step with a specific use case in mind. Would also create problems with the default for `tag` as it would become:
```
dockerComposeRun('sciencebeam-utils', './something.sh', true, commit)
dockerComposeRun('sciencebeam-utils', './something.sh', false, commit)
dockerComposeRun('sciencebeam-utils', './something.sh', true)
```
(because `commit`, having a default, needs to stay at the end)

If we then extended it:
```
dockerComposeRun('sciencebeam-utils', './something.sh', false, true, commit)
dockerComposeRun('sciencebeam-utils', './something.sh', true, false, commit)
```
will break all callers that pass 4 parameters, needing their Jenkinsfile to be updated. Some kind of enum would be helpful but with only 1 use case in mind it's difficult to design and when it changes needs the same BC break. That's a summary of why I am wary about complex parameter interfaces for Jenkins steps.

**Pull request merged**